Automattic/jetpack
# Instant Search: Integrate new Redux store with component tree

Follow-up for #18137.

#### Changes proposed in this Pull Request:
* Integrates the new Redux store with the component tree using React-Redux.
* Prunes the query string library to handle only getting and setting the query string. 
* Modify our Webpack configuration to enable aliasing React imports to `preact/compat`.
* Adds `redux` and `react-redux` as dependencies, which increases bundle bloat by 7.6 kB gzipped.
(Before this PR: 40.12 kB gzipped. After this PR: 47.59 kB gzipped.)

#### Does this pull request change what data or activity we track or use?
None.

#### Testing instructions:
* Apply this change to your Jetpack site with Jetpack Search.
* Try testing the following flows:
  1. Spawn the overlay by submitting something into a search input.
  2. Spawn the overlay by clicking on a filter that doesn't exist inside the overlay sidebar.
  3. Spawn the overlay by clicking on a filter that also exists inside the overlay sidebar.
  4. After spawning the overlay, try changing your search query.
  5. After spawning the overlay, try changing your sort type.
  6. After spawning the overlay, try changing your filters.
  7. Try opening a search URL from a new tab (e.g. `?s=home&category=uncategorized`).

For each of the flows, ensure that:
  - The search interface works as expected.
  - The address bar updates when the search query, sort, or filters are updated.

#### Proposed changelog entry for your changes:
* None.
